### PR TITLE
RTEMIS-179: Fixed all the lottery issues.

### PR DESCRIPTION
**Pull request by sandee-mahlawat-innoraft**
Ticket # : https://innoraft.atlassian.net/browse/RTEMIS-179

---
### What the PR does
`(Explain fixes, changes and impacts)`
- Design the buttons for lottery results.
- Modified the school pdf twig.
- Fixed all the lottery issues.

---
### What I have tested
`(Explain usecases you have tested locally)`
- Tested on different browsers and on different breakpoints.

### CODE REVIEW

File: modules/rte_mis_lottery/rte_mis_lottery.module [hunk 1]
```diff
@@ -70,3 +76,17 @@ function rte_mis_lottery_views_pre_render(ViewExecutable $view) {
...
+
+/**
+ * Implements hook_batch_alter().
+ */
+function rte_mis_lottery_batch_alter(&$batch) {
+  // Check if the operations is set.
+  $operations = $batch['sets'][0]['operations'];
+
+  // If the current batch is `lottery result`.
+  if (isset($operations) && isset($operations[0][1][0]) && $operations[0][1][0] == 'lottery_results') {
```
Comment (kunal-sin) [inline modules/rte_mis_lottery/rte_mis_lottery.module:88]: Please add `empty` check for $operations

File: modules/rte_mis_lottery/src/Services/RteLotteryHelper.php [hunk 2]
```diff
@@ -218,12 +218,14 @@ public function getStudentGroupBySchool() {
    *   Type of lottery.
    * @param mixed $academic_session
    *   Academic Session.
+   * @param array $ids
+   *   Student Ids.
    */
-  public function getLotteryResult($type, $academic_session) {
+  public function getLotteryResult($type, $academic_session, $ids = []) {
```
Comment (kunal-sin) [inline modules/rte_mis_lottery/src/Services/RteLotteryHelper.php:224]: Function param type missing for all the parameters.

Comment (RohanSingh-inno) [reply modules/rte_mis_lottery/src/Services/RteLotteryHelper.php]: okay

File: modules/rte_mis_lottery/rte_mis_lottery.module [hunk 3]
```diff
@@ -70,3 +76,17 @@ function rte_mis_lottery_views_pre_render(ViewExecutable $view) {
...
+
+/**
+ * Implements hook_batch_alter().
+ */
+function rte_mis_lottery_batch_alter(&$batch) {
+  // Check if the operations is set.
+  $operations = $batch['sets'][0]['operations'];
+
+  // If the current batch is `lottery result`.
+  if (!empty($operations) && !empty($operations[0][1]) && $operations[0][1][0] == 'lottery_results') {
```
Comment (kunal-sin) [inline modules/rte_mis_lottery/rte_mis_lottery.module:88]: @RohanSingh-inno The safe check if still missing.

Comment (RohanSingh-inno) [reply modules/rte_mis_lottery/rte_mis_lottery.module:88]: Fixed.

**Pull request merged**